src/mono/wasm/runtime/assets.ts

```diff
@@ -76,72 +126,65 @@ export async function mono_download_assets(): Promise<void> {
     }
 }
 
-export async function start_asset_download(asset: AssetEntry): Promise<AssetEntry | undefined> {
+export async function start_asset_download(asset: AssetEntryInternal, downloadData: boolean): Promise<ArrayBuffer | undefined> {
     try {
-        return await start_asset_download_throttle(asset);
+        return await start_asset_download_throttle(asset, downloadData);
     } catch (err: any) {
+        asset.internalPending = undefined;
```

Comment:
Are there cases (maybe node?) where we are not fetching from http* but from a filesystem? In that case do we have any need to retry?

Reply:
yeah, for filesystem access we shouldn't retry and should fail-fast. theoretically the config in browser could also be fetching from file:// or blob: